e3sm_diags/driver/utils/dataset_xr.py

```diff
@@ -385,10 +385,10 @@ def _get_climo_dataset(self, season: str) -> xr.Dataset:
         filepath = self._get_climo_filepath(season)
         ds = self._open_climo_dataset(filepath)
 
-        if self.var in ds.variables:
-            pass
-        elif self.var in self.derived_vars_map:
+        if self.var in self.derived_vars_map:
             ds = self._get_dataset_with_derived_climo_var(ds)
+        elif self.var in ds.data_vars.keys():
+            pass
         else:
```

Comment:
Hi @tomvothecoder I still think there is some change in derived var handling causing problems in my branch: https://github.com/E3SM-Project/e3sm_diags/pull/752#issuecomment-1928488372 But changing `dataset_xr.py` would impact all sets that has been refactored. The problems are not showing up for this set, because less dataset is used in zonal mean 2d. We can keep it as is, but I'm just making a note for this general issue.